server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenServices.java

```diff
@@ -489,7 +491,7 @@ private CompositeToken createCompositeToken(String tokenId,
 
     private static Map<String, Object> addRootClaimEntry(Map<String, Object> additionalRootClaims, String entry, String value) {
         Map<String, Object> claims = additionalRootClaims != null ? additionalRootClaims : new HashMap<>();
-        claims.put(entry, value);
+        claims.put(entry, CLIENT_AUTH_EMPTY.equals(value) ? CLIENT_AUTH_NONE : value);
```

Comment:
@peterhaochen47 here is the trick.

With this I set `client_auth_method=none` to the token, but internally we now that it was an empty secret and the deny of refresh - see line 327 - is only with none